src/nwb_datajoint/spikesorting/spikesorting_curation.py

```diff
@@ -1,61 +1,69 @@
 import json
 import os
 import shutil
+import time
 import uuid
 from pathlib import Path
+from typing import List
 from warnings import WarningMessage
 
 import datajoint as dj
+import numpy as np
 import spikeinterface as si
 import spikeinterface.toolkit as st
 
 from ..common.common_interval import IntervalList
 from ..common.common_nwbfile import AnalysisNwbfile
 from ..common.dj_helper_fn import fetch_nwb
+from .merged_sorting_extractor import MergedSortingExtractor
 from .spikesorting_recording import SpikeSortingRecording
 from .spikesorting_sorting import SpikeSorting
 
 schema = dj.schema('spikesorting_curation')
 
 
+def apply_merge_groups_to_sorting(sorting: si.BaseSorting, merge_groups: List[List[int]]):
+    # return a new sorting where the units are merged according to merge_groups
+    # merge_groups is a list of lists of unit_ids.
+    # for example: merge_groups = [[1, 2], [5, 8, 4]]]
+
+    return MergedSortingExtractor(parent_sorting=sorting, merge_groups=merge_groups)
+
+
 @schema
 class Curation(dj.Manual):
     definition = """
     # Stores each spike sorting; similar to IntervalList
-    curation_id: varchar(10)
+    curation_id: varchar(10) #A unique identifier for each curation entry for convenience
     -> SpikeSorting
     ---
+    curation_num = -1: int #
     parent_curation_id='': varchar(10)
     labels: blob # a dictionary of labels for the units
     merge_groups: blob # a list of merge groups for the units
     metrics: blob # a list of quality metrics for the units (if available)
     description='': varchar(1000) #optional description for this curated sort
+    time_of_creation: int   # in Unix time, to the nearest second
     """
 
     @staticmethod
-    def insert_curation(sorting_key: dict, parent_curation_id: str = 'original',
-                        labels=None, merge_groups=None, metrics=None,
-                        description=''):
-        """Given a SpikeSorting key and the `parent_sorting_id`
-        (and optional arguments) insert an entry into Curation
-
-        Parameters
-        ----------
-        sorting_key : dict
-            the key for the original SpikeSorting
-        parent_sorting_id : str
-            the id of the parent sorting (optional)
-        parent_sorting_id : str
-        param metrics : dict
-            computed metrics for sorting (optional)
-        description : str
-            text description of this sort (optional)
-        labels : dict
-            dictionary of labels for this sort (optional)
-
-        Returns
-        -------
-        sorting_id : str
+    def insert_curation(sorting_key: dict, parent_curation_id: int = -1, labels=None, merge_groups=None, metrics=None, description=''):
```

Comment:
This doesn't seem to be called automatically at the end of `make` of `SpikeSorting`. I think it should be called there because you always want to insert a `CurationTask` after spike sorting to be able to use it

Reply:
Now that I think about it, what happens if the user calls this function twice with the same key from `SpikeSorting`? given that a unique UUID is generated, wouldn't it insert two different entries rather than printing out a message saying it has already been inserted? that seems bad

Reply:
Just added this into SpikeSorting, but it's not yet tested.